x-pack/solutions/security/plugins/security_solution/public/asset_inventory/hooks/use_fetch_chart_data/use_fetch_chart_data.ts

```diff
@@ -28,14 +30,20 @@ export function useFetchChartData(options: UseTopAssetsOptions) {
     notifications: { toasts },
   } = useKibana().services;
 
+  const { dataView } = useDataViewContext();
+
+  const dataViewIndexPattern = useMemo(() => {
+    return dataView?.getIndexPattern();
+  }, [dataView]);
+
   return useQuery(
     [QUERY_KEY_CHART_DATA, { params: options }],
     async () => {
       const {
         rawResponse: { aggregations },
       } = await lastValueFrom(
         data.search.search<TopAssetsRequest, TopAssetsResponse>({
-          params: getTopAssetsQuery(options) as TopAssetsRequest['params'],
+          params: getTopAssetsQuery(options, dataViewIndexPattern) as TopAssetsRequest['params'],
```

Comment:
Can you also update the `enabled` on line `57` to consider the presence of the dataView before attempting to run the query? That will add an extra layer of checking to prevent unnecessary requests:

```tsx
enabled: options.enabled && !!dataViewIndexPattern,
```

As a context, we already have that on the `use_fetch_grid_data.ts`:

https://github.com/elastic/kibana/blob/22e6f78b843d8fc6a4504e3baae065340323f882/x-pack/solutions/security/plugins/security_solution/public/asset_inventory/hooks/use_fetch_grid_data.ts#L107